AK/Assertions.cpp

```diff
@@ -10,7 +10,7 @@
 #include <AK/StringBuilder.h>
 #include <AK/StringView.h>
 
-#if (defined(AK_OS_LINUX) && !defined(AK_OS_ANDROID)) || defined(AK_LIBC_GLIBC) || defined(AK_OS_BSD_GENERIC) || defined(AK_OS_SOLARIS) || defined(AK_OS_HAIKU)
+#if __has_include(<execinfo.h>)
 #    define EXECINFO_BACKTRACE
 #endif
 
```

Comment:
This actually isn't *quite* sufficient. In the two Android NDKs I have handy, 25.1.8937393 and 26.0.10404224, the contents of `<execinfo.h>` look like this:

```
#if __ANDROID_API__ >= 33
int backtrace(void* _Nonnull * _Nonnull buffer, int size) __INTRODUCED_IN(33);
... (other methods from this header)
#endif
```

So, in the check below for `__ANDROID_API__  >= 33` the `#else` case needs to `#undef EXECINFO_BACKTRACE`,

Reply:
It also makes me wonder if other vendors are this rude, to have a header exist but not provide any symbols (for this specific header). Though for android it *kind of* makes sense? I would just rather they complain at us with the availability macro (you know, like it's *designed* to do?) rather than #ifdef the symbols away.